databuilder/rcpai/base_data_loader.py

```diff
@@ -0,0 +1,135 @@
+import os
+import uuid
+
+from elasticsearch import Elasticsearch
+from pyhocon import ConfigFactory
+
+from databuilder.extractor.neo4j_extractor import Neo4jExtractor
+from databuilder.extractor.neo4j_es_last_updated_extractor import Neo4jEsLastUpdatedExtractor
+from databuilder.extractor.neo4j_search_data_extractor import Neo4jSearchDataExtractor
+from databuilder.job.job import DefaultJob
+from databuilder.loader.file_system_elasticsearch_json_loader import FSElasticsearchJSONLoader
+from databuilder.loader.file_system_neo4j_csv_loader import FsNeo4jCSVLoader
+from databuilder.publisher.elasticsearch_publisher import ElasticsearchPublisher
+from databuilder.publisher.neo4j_csv_publisher import Neo4jCsvPublisher
+from databuilder.task.task import DefaultTask
+from databuilder.transformer.base_transformer import NoopTransformer
+from databuilder.rcpai.base_arg_parser import Neo4JConf
+
+
+class BaseDataLoader(object):
+    base_dir: str = '/var/tmp/amundsen'
+    search_data_path: str = os.path.join(base_dir, 'search_data.json')
+    updated_data_dir: str = os.path.join(base_dir, 'last_updated_data')
+
+    def __init__(self, es_client: Elasticsearch, neo4j_conf: Neo4JConf):
+        self.es_client = es_client
+        self.neo4j_conf = neo4j_conf
+
+    def load(self, *args, **kwargs):
+        loading_job = self.create_extract_job(*args, **kwargs)
+        loading_job.launch()
+
+        self.create_last_updated_job().launch()
+
+        job_es_table = self.create_es_publisher_sample_job(
+            elasticsearch_index_alias='table_search_index',
+            elasticsearch_doc_type_key='table',
+            model_name='databuilder.models.table_elasticsearch_document.TableESDocument')
+        job_es_table.launch()
+
+    def create_extract_job(self, *args, **kwargs) -> DefaultJob:
```

Comment:
This is meant to be extended and implemented, as done so in YugabyteSQLLoader and MinioLoader. Otherwise it will raise an extension.